src/v/coproc/event_listener.cc

```diff
@@ -252,4 +175,35 @@ event_listener::poll_topic(model::record_batch_reader::data_t& events) {
                                  : ss::stop_iteration::no;
 };
 
+ss::future<>
+event_listener::process_events(event_batch events, model::offset last_offset) {
+    for (auto& [type, prepared_events] : events) {
+        auto it = _handlers.find(type);
+        if (it == _handlers.end()) {
+            vlog(
+              coproclog.warn,
+              "Can not find handler for coproc_type: {}",
+              coproc_type_as_string_view(type));
+            continue;
+        }
+
+        switch (type) {
+        case coproc_type::nodejs: {
+            try {
+                co_await it->second->process(std::move(prepared_events));
+            } catch (event_handler_exception& ex) {
+                _offset = last_offset;
+            }
+            break;
+        }
+        case coproc_type::data_policy: {
+            co_await it->second->process(std::move(prepared_events));
+            break;
+        }
+        default:
+            vassert(false, "Unexpected coproc_type");
```

Comment:
shouldn't crash here, just ignore the message. Crashing here would mean people would easily be able to force a crash just by sending a piece of data through the API that is improperly constructed, just log why exactly its unexpected and move on

Reply:
done